src/test/java/com/googleinterns/zoomtube/servlets/DiscussionServletTest.java

```diff
@@ -0,0 +1,205 @@
+    when(request.getParameter(DiscussionServlet.PARAM_LECTURE)).thenReturn(LECTURE_ID_STR);
+    StringWriter content = new StringWriter();
+    PrintWriter writer = new PrintWriter(content);
+    when(response.getWriter()).thenReturn(writer);
+
+    servlet.doGet(request, response);
+
+    verify(response).setContentType("application/json;");
+    List<Comment> comments = getCommentsFromJson(content.toString());
+    assertThat(comments.isEmpty()).isTrue();
+  }
+
+  @Test
+  public void doGet_returnsComments() throws ServletException, IOException {
+    when(request.getParameter(DiscussionServlet.PARAM_LECTURE)).thenReturn(LECTURE_ID_STR);
+    datastore.put(createTestCommentEntity(LECTURE_ID));
+    datastore.put(createTestCommentEntity(LECTURE_ID));
+    datastore.put(createTestCommentEntity(LECTURE_ID));
+    StringWriter content = new StringWriter();
+    PrintWriter writer = new PrintWriter(content);
+    when(response.getWriter()).thenReturn(writer);
+
+    servlet.doGet(request, response);
+
+    verify(response).setContentType("application/json;");
+    List<Comment> comments = getCommentsFromJson(content.toString());
+    assertThat(comments.size()).isEqualTo(3);
+  }
+
+  @Test
+  public void doGet_returnsCommentsForSpecificLecture() throws ServletException, IOException {
+    final int lectureA = 1;
+    final int lectureB = 2;
+    final int lectureC = 3;
+    // Looking for two comments under LECTURE_A.
+    when(request.getParameter(DiscussionServlet.PARAM_LECTURE))
+        .thenReturn(Integer.toString(lectureA));
+    datastore.put(createTestCommentEntity(lectureB));
+    datastore.put(createTestCommentEntity(lectureC));
+    datastore.put(createTestCommentEntity(lectureA)); // Add the first.
+    datastore.put(createTestCommentEntity(lectureB));
+    datastore.put(createTestCommentEntity(lectureB));
+    datastore.put(createTestCommentEntity(lectureC));
+    datastore.put(createTestCommentEntity(lectureA)); // Add the second.
+    datastore.put(createTestCommentEntity(lectureC));
+    StringWriter content = new StringWriter();
+    PrintWriter writer = new PrintWriter(content);
+    when(response.getWriter()).thenReturn(writer);
+
+    servlet.doGet(request, response);
+
+    verify(response).setContentType("application/json;");
+    List<Comment> comments = getCommentsFromJson(content.toString());
+    assertThat(comments.size()).isEqualTo(2);
+  }
+
+  private Entity createTestCommentEntity(int lectureId) {
+    Entity commentEntity = new Entity(Comment.ENTITY_KIND);
+    Key lecture = KeyFactory.createKey(/* kind= */ "Lecture", lectureId);
+    commentEntity.setProperty(Comment.PROP_LECTURE, lecture);
+    commentEntity.setProperty(Comment.PROP_PARENT, null);
+    commentEntity.setProperty(Comment.PROP_TIMESTAMP, new Date(0));
+    commentEntity.setProperty(Comment.PROP_AUTHOR, new User(DEFAULT_EMAIL, AUTH_DOMAIN));
+    commentEntity.setProperty(Comment.PROP_CONTENT, TEST_COMMENT_CONTENT);
```

Comment:
Is this data related to the other test which also uses TEST_COMMENT_CONTENT? I would suggest not using the same value as it adds confusion as to if they are related. If they are related then it should be more obvious. Eg, you can pass in this field.

Reply:
Same thing for DEFAULT_EMAIL and AUTH_DOMAIN

Reply:
I removed these constants as there were only used in one place and their values were never tested anywhere,